pyproject.toml

```diff
@@ -1,14 +1,93 @@
+name = "probnum"
+description = "Probabilistic Numerics in Python."
+readme = "README.md"
+requires-python = ">=3.8,<3.11"
+license = {file = "LICENSE.txt"}
+authors = [
+    {name = "Jonathan Wenger",   email = "jonathan.wenger@uni-tuebingen.de"},
+    {name = "Nicholas Kraemer",  email = "nicholas.kraemer@uni-tuebingen.de "},
+    {name = "Marvin Pfoertner",  email = "marvin.pfoertner@uni-tuebingen.de"},
+    {name = "Jonathan Schmidt",  email = "jonathan.schmidt@uni-tuebingen.de"},
+    {name = "Nathanael Bosch",   email = "nathanael.bosch@uni-tuebingen.de"},
+    {name = "Alexandra Gessner", email = "alexandra.gessner@uni-tuebingen.de"},
+    {name = "Toni Karvonen",     email = "toni.karvonen@helsinki.fi"},
+    {name = "Maren Mahsereci",   email = "maren.mahsereci@uni-tuebingen.de "},
+]
+maintainers = [
+    {name = "Jonathan Wenger",   email = "jonathan.wenger@uni-tuebingen.de"},
+    {name = "Nicholas Kraemer",  email = "nicholas.kraemer@uni-tuebingen.de "},
+    {name = "Marvin Pfoertner",  email = "marvin.pfoertner@uni-tuebingen.de"},
+    {name = "Maren Mahsereci",   email = "maren.mahsereci@uni-tuebingen.de "},
+    {name = "Jonathan Schmidt",  email = "jonathan.schmidt@uni-tuebingen.de"},
+]
+keywords = [
+    "probabilistic-numerics",
+    "machine-learning",
+    "numerical-methods",
+]
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+]
+dependencies = [
+    "numpy>=1.20",
+    "numpy>=1.21.3; python_version>='3.10'",
+    "scipy>=1.4",
+    "scipy>=1.8.0; python_version>='3.10'",
+]
+dynamic = ["version"]
+
+# TODO (pypa/setuptools#3221): Optional dependencies that reference one another
+# don't seem to be supported yet. Migrate the optional dependencies in `setup.py`
+# here, once this is supported.
+# [project.optional-dependencies]
+# full = [
+#     "%(jax)s",
+#     "%(zoo)s",
+#     "%(pls_calib)s",
+# ]
+
+[project.urls]
+github = "https://github.com/probabilistic-numerics/probnum"
+documentation = "https://probnum.readthedocs.io/en/stable/"
+benchmarks = "https://probabilistic-numerics.github.io/probnum-benchmarks/benchmarks/"
+
+################################################################################
+# PEP 518 Build System Configuration                                           #
+################################################################################
+
+# see https://peps.python.org/pep-0518/
 
 [build-system]
 requires = [
-    "setuptools>=42",
+    "setuptools>=61",
     "wheel",
     "setuptools_scm[toml]>=6.0",
 ]
 build-backend = "setuptools.build_meta"
 
+# see https://setuptools.pypa.io/en/stable/userguide/pyproject_config.html#setuptools-specific-configuration
+
+[tool.setuptools]
+platforms = [
+    "any",
+]
+zip-safe = false
+packages = ["probnum"]
```

Comment:
I specified `probnum` here manually instead of collecting it magically

Reply:
This means that any additional packages under `src` that we want installed must also be listed here in the future, which gives a very nice overview of the package contents.